packages/blockfrost/test/e2e/queryTransactions.test.ts

```diff
@@ -26,5 +26,114 @@ describe('blockfrostWalletProvider', () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('c09d19f5ac172e35dbeb7b279d54de73f7e997e49ca812e446fa362a43b71b58')
+      ]);
+      expect(tx.body.mint!).toBeDefined();
+    });
+    it('has StakeKeyRegistration cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('bcd165c882dc17a416bfef7053f0e1cfc3d715f8d7fc05a9803309f795878d9b')
+      ]);
+      expect(tx.body.certificates![0]).toEqual({
+        __typename: 'StakeKeyRegistration',
+        certIndex: 0,
+        rewardAccount: 'stake_test1upqykkjq3zhf4085s6n70w8cyp57dl87r0ezduv9rnnj2uqk5zmdv'
+      });
+    });
+    it('has StakeDelegation cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('bcd165c882dc17a416bfef7053f0e1cfc3d715f8d7fc05a9803309f795878d9b')
+      ]);
+      expect(tx.body.certificates![1]).toEqual({
+        __typename: 'StakeDelegation',
+        certIndex: 1,
+        delegationIndex: 1,
+        epoch: 182,
+        poolId: 'pool167u07rzwu6dr40hx2pr4vh592vxp4zen9ct2p3h84wzqzv6fkgv',
+        rewardAccount: 'stake_test1upqykkjq3zhf4085s6n70w8cyp57dl87r0ezduv9rnnj2uqk5zmdv'
+      });
+    });
+    it('has StakeKeyDereegistration cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('150a445f4d1f2e692791daec9c09f32d6c8c25a3f9ca6c7cf14ff8085375aaa0')
+      ]);
+      expect(tx.body.certificates![0]).toEqual({
+        __typename: 'StakeKeyDeregistration',
+        certIndex: 0,
+        rewardAccount: 'stake_test1uqe2twywhfjwt88ghas4sfgq7pp7m8wq64hlhz0vr4uhu2sj2tuzt'
+      });
+    });
+    it('has PoolRegistration cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('295d5e0f7ee182426eaeda8c9f1c63502c72cdf4afd6e0ee0f209adf94a614e7')
+      ]);
+      expect(tx.body.certificates![0]).toEqual({
+        __typename: 'PoolRegistration',
+        epoch: 76,
+        poolId: 'pool1y25deq9kldy9y9gfvrpw8zt05zsrfx84zjhugaxrx9ftvwdpua2',
+        poolParameters: null
+      });
+    });
+    it('has PoolRetirement cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('545ee7080a01aa085be01dffc073020be04ea3283b945c408c0830e9c4f8253c')
+      ]);
+      expect(tx.body.certificates![0]).toEqual({
+        __typename: 'PoolRetirement',
+        epoch: 80,
+        poolId: 'pool1ky9w02c4m4y842ygc0jyaf908wgllxv3cvpx42fd243f7uk664s'
+      });
+    });
+    it('has MoveInstantaneousRewards cert', async () => {
+      const [tx] = await walletProvider.queryTransactionsByHashes([
+        Cardano.TransactionId('24986c0cd3475419bfb44756c27a0e13a6354a4071153f76a78f9b2d1e596089')
+      ]);
+      expect(tx.body.certificates![0]).toEqual({
+        __typename: 'MoveInstantaneousRewards',
```

Comment:
This doesn't look right. `__typename` has different values ([`"MirCertificate"`](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Cardano/types/Certificate.ts#L9) in this case). Please debug this @iadmytro, [blockfrost code of mapping MIR certs](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/blockfrost/src/blockfrostWalletProvider.ts#L206) looks ok from first glance

Reply:
Updated tests with new __typename values